OptimizelySDK.Tests/NotificationTests/NotificationCenterTests.cs

```diff
@@ -0,0 +1,195 @@
+            Assert.AreEqual(3, NotificationCenter.NotificationsCount);
+        }
+
+        [Test]
+        public void TestAddSameNotificationListenerMultipleTimes()
+        {
+            var notificationTypeDecision = NotificationCenter.NotificationType.Decision;
+            NotificationCenter.AddNotification(notificationTypeDecision, TestNotificationCallbacks.TestDecisionCallback);
+
+            // Verify that adding same callback multiple times will gets failed.
+            Assert.AreEqual(-1, NotificationCenter.AddNotification(notificationTypeDecision,
+                TestNotificationCallbacks.TestDecisionCallback));
+            Assert.AreEqual(1, NotificationCenter.NotificationsCount);
+        }
+
+        [Test]
+        public void TestClearNotifications()
+        {
+            var notificationTypeDecision = NotificationCenter.NotificationType.Decision;
+            var notificationTypeTrack = NotificationCenter.NotificationType.Track;
+
+            // Add decision notifications.
+            NotificationCenter.AddNotification(notificationTypeDecision, TestNotificationCallbacks.TestDecisionCallback);
+            NotificationCenter.AddNotification(notificationTypeDecision, TestNotificationCallbacks.TestAnotherDecisionCallback);
+
+            // Add track notification.
+            NotificationCenter.AddNotification(notificationTypeTrack, TestNotificationCallbacks.TestTrackCallback);
+
+            // Verify that callbacks added successfully.
+            Assert.AreEqual(3, NotificationCenter.NotificationsCount);
+
+            // Verify that only decision callbacks are removed.
+            NotificationCenter.ClearNotifications(notificationTypeDecision);
+            Assert.AreEqual(1, NotificationCenter.NotificationsCount);
+        }
+
+        [Test]
+        public void TestClearAllNotifications()
+        {
+            var notificationTypeDecision = NotificationCenter.NotificationType.Decision;
+            var notificationTypeTrack = NotificationCenter.NotificationType.Track;
+
+            // Add decision notifications.
+            NotificationCenter.AddNotification(notificationTypeDecision, TestNotificationCallbacks.TestDecisionCallback);
+            NotificationCenter.AddNotification(notificationTypeDecision, TestNotificationCallbacks.TestAnotherDecisionCallback);
+
+            // Add track notification.
+            NotificationCenter.AddNotification(notificationTypeTrack, TestNotificationCallbacks.TestTrackCallback);
+
+            // Verify that callbacks added successfully.
+            Assert.AreEqual(3, NotificationCenter.NotificationsCount);
+
+            // Verify that ClearAllNotifications remove all the callbacks.
+            NotificationCenter.ClearAllNotifications();
+            Assert.AreEqual(0, NotificationCenter.NotificationsCount);
+        }
+
+        [Test]
+        public void TestFireNotifications()
+        {
```

Comment:
add one more test case when firenotifications is called when no notification is there. 
So have these scenarios. 
1 - ClearAllNotifications, firenotification for every type. and you must log that there is no notification registered.
2 - ClearNotification (notification type), fire notification for that notification type and log that there is no notification registered.
 @oakbani Please add in php as well.

Reply:
Good idea, will add that unit test.

Reply:
Added this case in this unit test.